tests/test_functions.py

```diff
@@ -27,6 +27,17 @@ def test_forward_hash():
     assert khmer.forward_hash('GGGG', 4) == 170
 
 
+def test_get_file_writer_fail():
+    somefile = utils.get_temp_filename("potato")
+    somefile = open(somefile, "w")
+
+    try:
+        get_file_writer(somefile, True, True)
+        assert False, "Shouldn't get to this"
+    except Exception as err:
```

Comment:
AssertionError is a subclass of exception, so you need to take a different approach here.  I would suggest setting a flag in place of the first assert, and then checking to make sure the flag *isn't* set.